src/Lean/Data/Lsp/Basic.lean

```diff
@@ -87,6 +99,22 @@ structure TextEdit where
   range : Range
   /-- The string to be inserted. For delete operations use an empty string. -/
   newText : String
+  /-- If this field is present and the editor supports it,
+  `newText` is ignored
+  and an interactive snippet edit is performed instead.
+
+  *Note* that this is a Lean-specific extension to the LSP standard,
+  so `newText` should still be set to a correct value
+  as fallback in case the editor does not support this feature.
+
+  *Note* that a snippet edit can always be performed as a normal text edit.
+  In VSCode, this will happen when no matching editor is open or when a workspace edit
+  contains snippet edits for multiple files. In that case only those that match the active editor
+  will be performed as snippet edits and the others as normal text edits. -/
+  /- NOTE: Similar functionality may be added to LSP in the future:
+  see [issue #592](https://github.com/microsoft/language-server-protocol/issues/592).
+  If such an addition occurs, this field should be deprecated. -/
+  leanExtSnippet? : Option SnippetString := none
```

Comment:
Multiple independent `*NOTE*`s, in different formats, makes me nervous here - it makes it sound like my files are getting deleted!

What about something like this?

```suggestion
  This is a Lean-specific extension to the LSP standard,
  so `newText` should still be set to a correct value
  as fallback in case the editor does not support this feature. Even editors
  that support snippets may not always use them; for instance, VS Code will
  only use snippets in the currently-active file when the language server
  provides edits to multiple files at once.
  -/
  /- NOTE: Similar functionality may be added to LSP in the future:
  see [issue #592](https://github.com/microsoft/language-server-protocol/issues/592).
  If such an addition occurs, this field should be deprecated. -/
  leanExtSnippet? : Option SnippetString := none
```

Reply:
Sure! Except I am now fully convinced of the utility of [semantic linebreaking](https://sembr.org/), so will format slightly differently.